blech_post_process.py

```diff
@@ -302,6 +328,358 @@
+        print()
+        print('======================================')
+        print()
+
+        # Iterate over electrodes and pull out spikes
+        # Get classifier probabilities for each spike and use only
+        # "good" spikes
+
+        # Print out selections
+        print(f'=== Processing Electrode {electrode_num:02} ===')
+
+        # Load data from the chosen electrode 
+        # We can pick any soluation, but need to know what
+        # solutions are present
+
+        (
+            spike_waveforms,
+            spike_times,
+            pca_slices,
+            energy,
+            amplitudes,
+            split_predictions,
+        ) = post_utils.load_data_from_disk(electrode_num, max_autosort_clusters) 
+
+        clf_data_paths = [
+                f'./spike_waveforms/electrode{electrode_num:02}/clf_prob.npy',
+                f'./spike_waveforms/electrode{electrode_num:02}/clf_pred.npy',
+                ]
+        clf_prob, clf_pred = [np.load(this_path) for this_path in clf_data_paths]
+
+        # If auto-clustering was done, data has already been trimmed
+        # Only clf_pred needs to be trimmed
+        clf_prob = clf_prob[clf_pred]
+        clf_pred = clf_pred[clf_pred]
+
+        ############################## 
+        # Calculate whether the cluster is a wanted_unit
+        # This will be useful for merging, so we only merge units
+
+        # Once selections have been made, save data
+        # Waveforms of originally chosen cluster
+        subcluster_inds = [np.where(split_predictions == this_split)[0] \
+                for this_split in np.unique(split_predictions)]
+        subcluster_waveforms = [spike_waveforms[this_inds] \
+                for this_inds in subcluster_inds]
+        subcluster_prob = [clf_prob[this_inds] \
+                for this_inds in subcluster_inds]
+        subcluster_times = [spike_times[this_inds] \
+                for this_inds in subcluster_inds]
+        mean_waveforms = [np.mean(this_waveform, axis = 0) for this_waveform in subcluster_waveforms]
+        std_waveforms = [np.std(this_waveform, axis = 0) for this_waveform in subcluster_waveforms]
+
+        # Check that the probability distributions are not uniform
+        # That indicates that the cluster is likely generated by noise
+
+        prob_dists = [np.histogram(this_prob, bins = 10, density = True)[0]
+                   for this_prob in subcluster_prob]
+
+        chi_out = [chisquare(this_dist) for this_dist in prob_dists]
+
+        chi_bool = [this_chi[1] < chi_square_alpha for this_chi in chi_out]
+        count_bool = [len(this_waveform) > count_threshold for this_waveform in subcluster_waveforms]
+
+        # To avoid confusion between index and cluster number,
+        # change fin_bool to dictionary because it gets used later
+        unique_clusters = np.unique(split_predictions)
+        fin_bool = np.logical_and(chi_bool, count_bool)
+        fin_bool_dict = dict(zip(unique_clusters, fin_bool))
```

Comment:
Instead of checking for "unit-ness" BEFORE merge, check after.
Because we can have 2 x 1000 waveforms merge into a proper unit

Reply:
Resolved in latest commit